src/main/scala/org/bitcoins/core/script/crypto/HashType.scala

```diff
@@ -1,47 +1,164 @@
 package org.bitcoins.core.script.crypto
 
-import org.bitcoins.core.util.{BitcoinSUtil}
+import org.bitcoins.core.number.Int32
+import org.bitcoins.core.util.Factory
 
 /**
  * Created by chris on 1/18/16.
  */
 sealed trait HashType {
+  def num : Int32
   def byte : Byte
-  def hex : String = BitcoinSUtil.encodeHex(byte)
+}
+
+object HashType extends Factory[HashType] {
+  def fromBytes(bytes : Seq[Byte]) : HashType = {
+    val num = Int32(bytes)
+    fromNumber(num)
+  }
+
```

Comment:
Let's create a `fromByte(byte: Byte)` function that takes in a single byte then calls `fromBytes` right away. This is nice for convenience so we don't have to wrap everything in a `Seq` to call fromBytes